### PR TITLE
[rllib] Support continuous action distributions in IMPALA/APPO

### LINKED ISSUE #4477: [rllib] Support all action spaces in IMPALA
Currently the IMPALA implementation is restricted to Discrete/MultiDiscrete, but this is just a implementation detail.

Cc @michaelzhiluo who has a patch for this

### PR DESCRIPTION
**Pull request by ericl**
## What do these changes do?

Replace the hardcoded softmax probability calculation with action_dist.logp() in vtrace.py. This should be enough to support envs with arbitrary action spaces.

- [x] Test Breakout hasn't regressed
- [x] test on pendulum

cc @stefanpantic @michaelzhiluo 

## Related issue number

Closes https://github.com/ray-project/ray/issues/4477

## Linter

- [x] I've run `scripts/format.sh` to lint the changes in this PR.

### CODE REVIEW

**Comment by ericl** [comment]
cc @joneswong could you take a look?

File: python/ray/rllib/agents/impala/vtrace_policy_graph.py [hunk 1]
```diff
@@ -140,30 +142,28 @@ def __init__(self,
...
         elif isinstance(action_space, gym.spaces.multi_discrete.MultiDiscrete):
             is_multidiscrete = True
-            actions_shape = [None, len(action_space.nvec)]
             output_hidden_shape = action_space.nvec.astype(np.int32)
         else:
-            raise UnsupportedSpaceException(
-                "Action space {} is not supported for IMPALA.".format(
-                    action_space))
+            is_multidiscrete = False
+            output_hidden_shape = 1
```
Comment (joneswong) [inline python/ray/rllib/agents/impala/vtrace_policy_graph.py]: why does `output_hidden_shape` be 1 in this branch

**Review by joneswong — APPROVED** [review]
looks good

File: python/ray/rllib/agents/impala/vtrace_policy_graph.py (hunk 1 above)
Comment (ericl) [reply python/ray/rllib/agents/impala/vtrace_policy_graph.py]: I believe this is the equivalent of `[n]` in the normal Discrete case above, which doesn't actually cause any splitting to happen below (splitting is only for multi-discrete).

**Pull request merged**